modules/order/src/AvailabilityOrderRefreshProcessor.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\commerce_order;
+
+use Drupal\commerce\AvailabilityManagerInterface;
+use Drupal\commerce_order\Entity\OrderInterface;
+
+/**
+ * Provides an order refresh processor for availability.
+ */
+class AvailabilityOrderRefreshProcessor implements OrderRefreshProcessorInterface {
+
+  /**
+   * The availability manager.
+   *
+   * @var \Drupal\commerce\AvailabilityManagerInterface
+   */
+  protected $availabilityManager;
+
+  /**
+   * Creates a new AvailabilityOrderRefreshProcessor object.
+   *
+   * @param \Drupal\commerce\AvailabilityManagerInterface $availability_manager
+   *   The availability manager.
+   */
+  public function __construct(AvailabilityManagerInterface $availability_manager) {
+    $this->availabilityManager = $availability_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function refresh(OrderInterface $order) {
+    foreach ($order->getLineItems() as $line_item) {
+      $available = $this->availabilityManager->check($line_item->getPurchasedEntity(), $line_item->getQuantity());
+
```

Comment:
What would the message be? Also I don't know where in 1.x we'd have something like this. 

In http://cgit.drupalcode.org/commerce/tree/modules/cart/commerce_cart.module#n1114 we removed line items with `null` price values, but this was silent as well.